From 3ef92719e7c374bb584d4ec274e86bc4408794c3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:18 -0400
Subject: [tools] lib/traceevent: Use global QUIET_LINK build output

Message-id: <1408953210-26343-191-git-send-email-jolsa@redhat.com>
Patchwork-id: 88902
O-Subject: [PATCH RHEL7.1 BZ1133083 190/322] tools lib traceevent: Use global QUIET_LINK build output
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit e6262e23134f575ae07ecc25bb69ddb63caebc95
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Dec 19 14:42:04 2013 +0100

description
===========
Using global QUIET_LINK build output variable and getting rid of local
print_static_lib_build, print_plugin_build and print_shared_lib_compile.

We no longer distinguish between shared and static library in the build
message. It's differenced by the built file suffix, like:

  $ make
    ...
    LINK     libtraceevent.a
    LINK     libtraceevent.so
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/Makefile b/tools/lib/traceevent/Makefile
index 24203cc..51be8ab 100644
--- a/tools/lib/traceevent/Makefile
+++ b/tools/lib/traceevent/Makefile
@@ -152,14 +152,9 @@ override CFLAGS += $(udis86-flags) -D_GNU_SOURCE
 
 ifeq ($(VERBOSE),1)
   Q =
-  print_shared_lib_compile =
-  print_plugin_build =
   print_install =
 else
   Q = @
-  print_shared_lib_compile = echo '  BUILD    SHARED LIB '$(OBJ);
-  print_plugin_build =  echo '  BUILD    PLUGIN '$(OBJ);
-  print_static_lib_build = echo '  BUILD    STATIC LIB '$(OBJ);
   print_install =  echo '  INSTALL  '$1;
 endif
 
@@ -216,10 +211,10 @@ all: all_cmd
 all_cmd: $(CMD_TARGETS)
 
 libtraceevent.so: $(PEVENT_LIB_OBJS)
- $(Q)$(do_compile_shared_library)
+ $(QUIET_LINK)$(CC) --shared $^ -o $@
 
 libtraceevent.a: $(PEVENT_LIB_OBJS)
- $(Q)$(do_build_static_lib)
+ $(QUIET_LINK)$(RM) $@; $(AR) rcs $@ $^
 
 plugins: $(PLUGINS)
 
@@ -230,7 +225,7 @@ $(PLUGIN_OBJS): %.o : $(src)/%.c
  $(QUIET_CC_FPIC)$(CC) -c $(CFLAGS) -fPIC -o $@ $<
 
 $(PLUGINS): %.so: %.o
- $(Q)$(do_plugin_build)
+ $(QUIET_LINK)$(CC) $(CFLAGS) -shared -nostartfiles -o $@ $<
 
 define make_version.h
  (echo '/* This file is automatically generated. Do not modify. */';  \
-- 
1.7.1